src/wallet/wallet.cpp

```diff
@@ -3630,6 +3628,17 @@ DBErrors CWallet::LoadWallet(bool& fFirstRunRet)
         }
     }
 
+    {
+        LOCK2(cs_main, cs_wallet);
+        for (auto& pair : mapWallet) {
+            for(int i = 0; i < pair.second.vout.size(); ++i) {
+                if (IsMine(pair.second.vout[i]) && !IsSpent(pair.first, i)) {
+                    setWalletUTXO.insert(COutPoint(pair.first, i));
```

Comment:
I'm wondering if this is not better placed in the first if branch of AddToWallet? Would result having setWalletUTXO manipulation in only one place.
https://github.com/dashpay/dash/pull/1655/files#diff-b2bb174788c7409b671c46ccc86034bdR884

Reply:
Yes, that was the way I implemented it first. But then I thought that (even though it's nice to have all related logic in on place) this approach is kind of suboptimal because when you load some old tx you insert utxo records, and if later you load spending txes for some/all of its outputs you have to remove them. However, if you do this when all txes are already loaded you only have to insert utxos once and there is no remove operation, so this way you skip all useless add/remove (and for huge wallets there could be a lot of them).

Reply:
Ah ok, got it. Then I'm fine with this.